### ISSUE TITLE
No "Neighbouring Cells" displayed under "Current Threat Level" 

### ISSUE THREAD

**Issue by pyr8** [SecUpwN/Android-IMSI-Catcher-Detector#630]
No "Neighbouring Cells" displayed under "Current Threat Level"

**Comment by agilob** [comment]
Yeah, I noticed the same recently, what device and ROM (version) do you have?
It worked on my S3 with stack android, but when I switched to CM AIMSICD became unusable, patch should be easy to make, but want to make sure it's not device or version dependent.

**Comment by pyr8** [comment]
Running latest nightly of cyanogenmod on Galaxy S4.

Yeah, I noticed the same recently, what device and ROM (version) do you
have?
It worked on my S3 with stack android, but when I switched to CM AIMSICD
became unusable, patch should be easy to make, but want to make sure it's
not device or version dependent.

—
Reply to this email directly or view it on GitHub
<https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/issues/630#issuecomment-150116780>
.

**Comment by agilob** [comment]
I think CM team removed one class in 10.2 release that provided required data. No proof, just speculations.

**Comment by pyr8** [comment]
???

**Comment by smarek** [comment]
@agilob could you please look up what class should have been removed? Curious what this speculation is based on.

**Comment by lop1** [comment]
Same thing here with my galaxy s4 mini stock rom 4.4.2

**Comment by Tonat-Systems** [comment]
Same issue with galaxy S3 on CM 12.1

**Comment by SecUpwN** [comment]
@larsgrefer, is https://github.com/SecUpwN/Android-IMSI-Catcher-Detector/issues/630#issuecomment-150170679 really the case? How can we get back hands on that important data?

**Comment by ProGamerGov** [comment]
I am running stock Android and have the same issue. It's a bug in the app, not because of CM.

**Comment by fabifri** [comment]
I am using CyanogenMod 12.1 (Android 5.1.1) on OnePlus Two. Telefon/SIM-Details showing fine, but no neighboring Cells at Current Threat Level. OpenCellID API Key is good and data downloaded.

If a developer needs a OnePlus Two to investigate on the issue - it's lying around here, so i can lend it or hand it over...

**Comment by SecUpwN** [comment]
Good morning @fabifri, welcome to our project and thanks for offering your device for testing! Please keep it and continue testing, a fresh version of our app will be released by the end of this week.

To everyone reading this Issue: Which **network type** do you use? Keep in mind that our app currently works best when locked on 3G. If you use LTE, our app will not throw an error, but will get no data (yet).

**Comment by agilob** [comment]
>Telefon/SIM-Details showing fine, but no neighboring Cells at Current Threat Level

That's device-dependent. On some you will get neighbouring cells, on some you won't.

**Comment by fabifri** [comment]
I am only using 3G (and sometimes 2G Networks).
Thanks for your Answers. Yeah, i'll continue testing - of course. ;-)

So, the Detection / Thread-Level-Display does still work, even if i can't see any Neighboring Cell Data?

**Comment by agilob** [comment]
no...

**Comment by davidak** [comment]
Same Problem on a Fairphone 2 with Fairphone OS 1.2.8 (Android 5.1), also locked to 3G.

**Comment by SecUpwN** [comment]
Folks, what we need is **LOGFILES**. Otherwise our developers will not be able to trace down the Issue. So to everyone wanting to help solving this: Please be so kind and record a full logcat with **working neighboring cells**, then do another one **without neighboring cells**. Detailled instructions can be found [here](https://github.com/CellularPrivacy/Android-IMSI-Catcher-Detector/blob/development/.github/CONTRIBUTING.md#debugging). Please remember to anonymize your logs by removing `CID`, `LAC` and othe rpersonla info and also to use [PASTEBIN](https://defuse.ca/pastebin.htm). Thank you.

**Comment by agilob** [comment]
@SecUpwN List of neighbouring cells is implemented correctly (still there is place for improvements), if there is no NC it means:
1. device doesn't support it, eg. most Samsung, some LGs doesn't support it at all
2. ROM doesn't support it, some ROMs removed that feature
3. You're using connection that doesn't have neighbouring cells like CDMA

It's normal, stop confusing people.

**Comment by SecUpwN** [comment]
>1. device doesn't support it, eg. most Samsung, some LGs doesn't support it at all
>2. ROM doesn't support it, some ROMs removed that feature

That is why we need #181 during which #269 needs to be executed to make that clear to our users.

**Comment by ProGamerGov** [comment]
On an older version of the app, I could see neighbouring cells. Not sure exactly what version changed that, but I can check if it would help fix this issue for some of the users.
